beacon_node/eth1/src/http.rs

```diff
@@ -48,6 +48,30 @@ pub enum BlockQuery {
     Latest,
 }
 
+/// Represents an error received from a remote procecdure call.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum RpcError {
+    NoResultField,
+    InvalidJson,
+    Error(String),
+}
+
+impl From<serde_json::Error> for RpcError {
+    fn from(_value: serde_json::Error) -> Self {
+        RpcError::InvalidJson
+    }
+}
+
+impl From<RpcError> for String {
+    fn from(value: RpcError) -> Self {
+        match value {
+            RpcError::NoResultField => "No JSON error returned".into(),
```

Comment:
This would probably be more helpful as somthing like: "No result field in JSON response"